sam/metadata.go

```diff
@@ -87,9 +87,11 @@ func ParseHeaderText(h Header) Header {
 	if h.Text != nil {
 		h.Metadata.AllTags, h.Metadata.Comments = parseTagsAndComments(h.Text)
 		h.Chroms = getChromInfo(h.Metadata.AllTags)
-		h.Metadata.Version = getVersion(h.Metadata.AllTags)
-		h.Metadata.SortOrder = getSortOrder(h.Metadata.AllTags)
-		h.Metadata.Grouping = getGrouping(h.Metadata.AllTags)
+		if _, ok := h.Metadata.AllTags[[2]byte{'H','D'}]; ok {
```

Comment:
Previous version of this metadata function required the header to start with @HD. However, this field is optional, so that has been corrected here.